api/pkg/apis/v1alpha1/providers/stage/script/script_test.go

```diff
@@ -61,3 +62,17 @@ func TestShellScriptOnline(t *testing.T) {
 	assert.Nil(t, err)
 	os.Remove("staging/go1.21.6.src.tar.gz")
 }
+
+func TestShellScriptNotFoundOnline(t *testing.T) {
+	provider := ScriptStageProvider{}
+	err := provider.Init(ScriptStageProviderConfig{
+		Name:          "test",
+		Script:        "test.ps1",
+		ScriptEngine:  "powershell",
+		ScriptFolder:  "https://bing.com",
+		StagingFolder: "staging",
+	})
+	assert.NotNil(t, err)
+	assert.IsType(t, v1alpha2.COAError{}, err)
+	assert.Contains(t, err.Error(), "failed to download script because of")
```

Comment:
What about checking statusCode?

Reply:
Since we can only know it's not `http.StatusOk`, I used NotEqual to checked it in the update.